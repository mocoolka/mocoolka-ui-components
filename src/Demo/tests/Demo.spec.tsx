import * as React from 'react';
import  enzyme from 'enzyme';
import  Adapter from 'enzyme-adapter-react-16';
import {App,Demo} from '../../'
import { shallow as enzymeShallow , EnzymeSelector } from 'enzyme';
import renderer from 'react-test-renderer';
enzyme.configure({ adapter: new Adapter() });
function shallowRecursively(wrapper, selector, { context, ...other }) {
    if (wrapper.isEmptyRender() || typeof wrapper.getElement().type === 'string') {
        return wrapper;
    }

    let newContext = context;

    const instance = wrapper.root().instance();
    // The instance can be null with a stateless functional component and react >= 16.
    if (instance && instance.getChildContext) {
        newContext = {
            ...context,
            ...instance.getChildContext(),
        };
    }

    const nextWrapper = wrapper.shallow({ context: newContext, ...other });

    if (selector && wrapper.is(selector)) {
        return nextWrapper;
    }

    return shallowRecursively(nextWrapper, selector, { context: newContext });
}

export  function until(selector, options:any = {}) {
    return this.single('until', () => shallowRecursively(this, selector, options));
}

// Generate an enhanced shallow function.
export default function createShallow(options1: any = {}) {
    const { shallow = enzymeShallow, dive = false, untilSelector = false, ...other1 } = options1;

    const shallowWithContext = function shallowWithContext(
        node: any,
        options2: any = {},
    ) {
        const options = {
            ...other1,
            ...options2,
            context: {
                ...other1.context,
                ...options2.context,
            },
        };

        const wrapper = shallow(node, options);

        if (dive) {
            return wrapper.dive();
        }

        if (untilSelector) {
            return until.call(wrapper, untilSelector, options);
        }

        return wrapper;
    };

    return shallowWithContext;
}
describe('<AppBar />', () => {
    let shallow= createShallow({ dive: true });
    let classes;

/*     it('should render a Paper component', () => {
        const wrapper = shallow(<App title='test'><Demo js="aaa" githubLocation="mocoolka" raw="test"/></App>);
        expect(wrapper).toEqual('')
    }); */
    it('should render a Paper component', () => {
        const tree = renderer.create(
            <App.App title='test'><Demo.Demo js="aaa" githubLocation="mocoolka" raw="test"/></App.App>,
          ).toJSON();
          
          // expect(tree).toMatchSnapshot();
    });

});
