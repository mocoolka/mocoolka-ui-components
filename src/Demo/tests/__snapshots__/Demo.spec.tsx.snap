// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBar /> should render a Paper component 1`] = `
<div
  className="sc-tilXH bfFGhT"
>
  <div
    className="sc-cHGsZl jNSBpC"
  >
    <div
      className="sc-kgAjT lmhSfy"
    >
      <form
        action="https://codesandbox.io/api/v1/sandboxes/define"
        method="post"
        target="_blank"
      >
        <input
          name="parameters"
          type="hidden"
          value=""
        />
      </form>
      <a
        className="sc-fAjcbJ eYmJjZ"
        href="mocoolka"
        onClick={[Function]}
      >
        <div
          className="sc-kjoXOD bEIIJO"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M12,4.5C7,4.5,2.7,7.6,1,12c1.7,4.4,6,7.5,11,7.5s9.3-3.1,11-7.5C21.3,7.6,17,4.5,12,
4.5z M12,17c-2.8,0-5-2.2-5-5
s2.2-5,5-5s5,2.2,5,5S14.8,17,12,17z M12,9c-1.7,0-3,1.3-3,3s1.3,3,3,3s3-1.3,3-3S13.7,9,12,9z"
              />
            </g>
          </svg>
        </div>
      </a>
      <button
        className="sc-fAjcbJ eYmJjZ"
        onClick={[Function]}
      >
        <div
          className="sc-kjoXOD bEIIJO"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <polygon
              points="9,19.4 3.3,13.7 4.7,12.3 9,16.6 19.8,5.8 21.2,7.2"
            />
          </svg>
        </div>
      </button>
      <button
        className="sc-fAjcbJ eYmJjZ"
        onClick={[Function]}
      >
        <div
          className="sc-kjoXOD bEIIJO"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
    <div
      className="sc-gPEVay hHStvP"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<pre><code class=\\"language-jsx\\">test</code></pre>
",
        }
      }
    />
    <div
      className="sc-TOsTZ gPbTSw"
    >
      <aaa />
    </div>
  </div>
  <div
    className="sc-ifAKCX kFngQp"
    onClick={[Function]}
  />
</div>
`;
